composer.lock

```diff
@@ -4127,97 +4129,31 @@
             ],
             "time": "2023-01-26T09:26:14+00:00"
         },
-        {
-            "name": "symfony/process",
-            "version": "v5.4.28",
-            "source": {
-                "type": "git",
-                "url": "https://github.com/symfony/process.git",
-                "reference": "45261e1fccad1b5447a8d7a8e67aa7b4a9798b7b"
-            },
-            "dist": {
-                "type": "zip",
-                "url": "https://api.github.com/repos/symfony/process/zipball/45261e1fccad1b5447a8d7a8e67aa7b4a9798b7b",
-                "reference": "45261e1fccad1b5447a8d7a8e67aa7b4a9798b7b",
-                "shasum": ""
-            },
-            "require": {
-                "php": ">=7.2.5",
-                "symfony/polyfill-php80": "^1.16"
-            },
-            "type": "library",
-            "autoload": {
-                "psr-4": {
-                    "Symfony\\Component\\Process\\": ""
-                },
-                "exclude-from-classmap": [
-                    "/Tests/"
-                ]
-            },
-            "notification-url": "https://packagist.org/downloads/",
-            "license": [
-                "MIT"
-            ],
-            "authors": [
-                {
-                    "name": "Fabien Potencier",
-                    "email": "fabien@symfony.com"
-                },
-                {
-                    "name": "Symfony Community",
-                    "homepage": "https://symfony.com/contributors"
-                }
-            ],
-            "description": "Executes commands in sub-processes",
-            "homepage": "https://symfony.com",
-            "support": {
-                "source": "https://github.com/symfony/process/tree/v5.4.28"
-            },
-            "funding": [
-                {
-                    "url": "https://symfony.com/sponsor",
-                    "type": "custom"
-                },
-                {
-                    "url": "https://github.com/fabpot",
-                    "type": "github"
-                },
-                {
-                    "url": "https://tidelift.com/funding/github/packagist/symfony/symfony",
-                    "type": "tidelift"
-                }
-            ],
-            "time": "2023-08-07T10:36:04+00:00"
-        },
         {
             "name": "symfony/service-contracts",
-            "version": "v2.5.2",
+            "version": "v3.4.1",
```

Comment:
You can't simply update these dependencies :)

Reply:
Why? There is no usage in the codebase, or is it? 

So this became transitive dependency after doing `composer remove symfony/service-contracts` (as it is still needed by some other package). So it got upgraded to latest version by composer (as any other transitive dependency here).

Reply:
Ok, you [partially explained it here](https://github.com/ondrejmirtes/simple-downgrader/issues/5#issuecomment-1961364675)

Reply:
> So this became transitive dependency

This is in composer.json `require` for the same reason jetbrains/phpstorm-stubs is there. To force a specific version.

You worked around it just by ignoring it, but I think that composer-dependency-analyser shouldn't mark dependencies in composer.json as unused if they're already a transitive dependency.

Reply:
> composer-dependency-analyser shouldn't mark dependencies in composer.json as unused if they're already a transitive dependency

This is hard to say in general. Currently, any dependency without a usage is marked as unused. I'd rather keep it as-is and let maintainers decide if they need it in composer.json (due to some version pinning like here) or they just forgot it there. Ignoring the error for specific packages is super easy and when commented, it documents the purpose why it is there even when unused.

Reply:
Ok, I removed only things that didnt break CI. 

So if your coverage is solid, upgrading `symfony/service-contracts` by removing direct dependency is safe here.